lib/spack/spack/test/ci.py

```diff
@@ -25,8 +29,146 @@ def repro_dir(tmp_path):
         yield result
 
 
-def test_urlencode_string():
-    assert ci._url_encode_string("Spack Test Project") == "Spack+Test+Project"
+def test_pipeline_dag(config, tmpdir):
+    """Make sure we achieve the best possible staging for the following
+spec DAG::
+
+        a
+       /|
+      c b
+        |\
+        e d
```

Comment:
This is not an improvement, because all breadth-first orders from `a` are topologically ordered. You have to create a dag structure where there are two paths from b to f with distinct length.